### PR TITLE
fuzz: Cleanup muhash fuzz target

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Added a commit to speed up the target, because it is by far the slowest one

**Review by fjahr — COMMENTED** [review]
Concept ACK

I think this can be simplified further to use only one `CallOneOf` but still achieve the same thing: https://github.com/fjahr/bitcoin/commit/2ff513b51bbb870611a7be3ac640507ade299370. I also added some more comments there and used a constant for the initial state hash which should help the speed some more.

File: src/test/fuzz/muhash.cpp [hunk 1]
```diff
@@ -12,52 +12,60 @@
...
+        fuzzed_data_provider,
+        [&] {
+            // Test that MuHash result is consistent independent of order of operations
+            uint256 out;
+            muhash.Finalize(out);
+            CallOneOf(
+                fuzzed_data_provider,
+                [&] {
+                    MuHash3072 muhash_inverse_order;
+                    muhash_inverse_order = MuHash3072();
```
[the remark this answers is not in the document]
Comment (MarcoFalke) [reply src/test/fuzz/muhash.cpp]: restored original code

**Comment by MarcoFalke** [comment]
> I think this can be simplified further to use only one CallOneOf but still achieve the same thing: fjahr@2ff513b. I also added some more comments there and used a constant for the initial state hash which should help the speed some more.

Thanks for the 2am review and code suggestion. Took ~your branch and squashed into one commit.

**Comment by MarcoFalke** [comment]
Speedup: Locally it now takes 29% of the original time

**Comment by MarcoFalke** [comment]
CI Before: 2654 s https://cirrus-ci.com/task/4629294711308288?logs=ci#L4695

CI After: 1011 s https://cirrus-ci.com/task/6682329813876736?logs=ci#L4189

**Pull request merged**